b2sdk/sync/folder.py

```diff
@@ -282,12 +280,8 @@ def _handle_non_unicode_file_name(cls, name):
         names.
         """
         # if it's all ascii, allow it
-        if six.PY2:
-            if all(ord(c) <= 127 for c in name):
-                return name
-        else:
-            if all(b <= 127 for b in name):
-                return name
+        if all(b <= 127 for b in name):
```

Comment:
by the way, isn't there a faster way to do this using `str.encode('ascii')` and catching an exception?

Reply:
Trying to decode bytes is a lot slower. This method should not be called if `self.root` is `str`. In my opinion, we should prevent using bytes for paths and raise an exception in such case

Reply:
yeah I think we should move it all to pathlib in v2 and leave wrappers for strings for v1. I updated the ticket with the list of things to consider for apiver v2.